server/mldb_server.cc

```diff
@@ -498,9 +498,11 @@ getPackageDocumentationPath(const Package & package) const
     // always be provided by the plugin "pro", but this is not
     // by any means guaranteed.
 
+    auto * rawBaseUrl = secure_getenv("HTTP_BASE_URL");
```

Comment:
This should be passed as a command line option and put as a member variable of MldbServer, not magically brought in from the environment IMO.  As for the performance aspects, I don't think they are worrisome, especially if it's stored in a local variable.